opa/opa.go

```diff
@@ -96,8 +98,30 @@ func (opa *client) InitializePolicy(policy string) ClientError {
 	}
 
 	if !exists {
-		// TODO: fetch rules from ES
-		return NewClientError("policy does not exist", OpaClientErrorTypePolicyNotFound, nil)
+		err := opa.loadPolicy(policy, policyData)
+		if err != nil {
+			return NewClientError(err.Error(), OpaClientErrorTypeLoadPolicy, err)
+		}
+	}
+
+	return nil
+}
+
+func (opa *client) loadPolicy(policy string, policyData string) error {
+	log := opa.logger.Named("Load Policy")
+	client := &http.Client{}
+	req, err := http.NewRequest(http.MethodPut, opa.getURL(fmt.Sprintf("v1/policies/%s", policy)), strings.NewReader(policyData))
+	if err != nil {
+		return fmt.Errorf("failed to generate policy request")
+	}
+	response, err := client.Do(req)
+	if err != nil {
+		log.Error("load policy error", zap.Error(err), zap.String("input", string(policyData)))
+		return fmt.Errorf("failed to load the policy in opa")
+	}
+
+	if response.StatusCode == http.StatusOK {
```

Comment:
Should there be an else here for non-200 responses?

Reply:
I can add one, but i don't even know how to get a non 200 response without also getting an error. Maybe i just remove this line @alexashley ??

Reply:
I think it makes sense to check the status code, even if OPA won't ever spit out anything other than 200, there could be a proxy or something in between Rode and OPA that could return non-200 (like a 503 if OPA was down).